### PR TITLE
Reformated all the code.

### PR DESCRIPTION
**Pull request by Jiwan**
@aexg what is your opinion on these reformatting rules? Any suggestions?

### CODE REVIEW

File: utils.hpp [hunk 1]
```diff
@@ -55,16 +55,7 @@ constexpr auto digits_amount(int x)
...
-        (x < 100 ? 2 :   
-        (x < 1000 ? 3 :   
-        (x < 10000 ? 4 :   
-        (x < 100000 ? 5 :   
-        (x < 1000000 ? 6 :   
-        (x < 10000000 ? 7 :  
-        (x < 100000000 ? 8 :  
-        (x < 1000000000 ? 9 :  
-        10)))))))));  
+    return (x < 10 ? 1 : (x < 100 ? 2 : (x < 1000 ? 3 : (x < 10000 ? 4 : (x < 100000 ? 5 : (x < 1000000 ? 6 : (x < 10000000 ? 7 : (x < 100000000 ? 8 : (x < 1000000000 ? 9 : 10)))))))));
```
Comment (Jiwan) [inline utils.hpp:58]: I could probably disable clang-tidy just for these lignes.

File: constexpr_string.hpp [hunk 2]
```diff
@@ -17,35 +17,39 @@ class constexpr_string
     using const_iterator = const char*;
 
 public:
-    constexpr constexpr_string(): data_{}, size_{N}
```
Comment (aexg) [inline constexpr_string.hpp:20]: I liked short constructor initializers list on same line as constructor, it also complied with Google style that we use at work, I'll see on Monday our file at work!
https://google.github.io/styleguide/cppguide.html#Constructor_Initializer_Lists

File: game_engine.hpp [hunk 3]
```diff
@@ -3,16 +3,16 @@
 
 #include <cstddef>
 #include <optional>
-#include <variant>
 #include <tuple>
+#include <variant>
```
Comment (aexg) [inline game_engine.hpp:8]: That reminds me to remove "<tuple>" that I added for BoardExtended that is now a struct.

File: performance_test/measure_time.py [hunk 4]
```diff
@@ -30,8 +30,8 @@ def measure(self, game_field, kbdinput):
         shutil.copy('./performance_test/' + game_field, 'current_state.txt')
         print('field = {:<15s} key = {:5s}    measuring '.format(
             game_field, kbdinput),
-              end='',
-              flush=True)
+            end='',
+            flush=True)
```
Comment (aexg) [inline performance_test/measure_time.py:34]: I check PEP8 compliance with `flake8` before committing, I still missed some spaces on 2 lines that autopep rectified. I learned the lesson and installed autopep now :-)

File: utils.hpp (hunk 1 above)
Comment (aexg) [reply utils.hpp:58]: I agree, the original layout is more readable.

**Review by aexg — APPROVED** [review]
It looks good, ready for merge!

**Review by aexg — APPROVED** [review]
I thought I already reviewed and approved, still see "review required, merge blocked". I approved from this link too! (or probably 05:42 after Saturday partying is inefficient time to interact with github?)

**Pull request merged**